composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt

```diff
@@ -149,6 +150,20 @@ class Engine(
         proxy = preferences.proxy,
     )
 
+    private fun maxRuntime(
+        taskOrigin: TaskOrigin,
+        netTest: NetTest,
+        preferences: EnginePreferences,
+    ) = if (taskOrigin == TaskOrigin.AutoRun) {
+        MAX_RUNTIME_DISABLED
+    } else if (netTest.callCheckIn == true) {
+        preferences.maxRuntime?.inWholeSeconds?.toInt()?.let { maxRuntimePreference ->
+            if (maxRuntimePreference > 0) 30 + maxRuntimePreference else MAX_RUNTIME_DISABLED
```

Comment:
Why are we adding 30 seconds to the preference time?

Reply:
legacy migration

Reply:
https://github.com/ooni/probe-android/blob/master/app/src/main/java/org/openobservatory/ooniprobe/test/test/WebConnectivity.java#L50-L60